### PR TITLE
API-2532: adding JSON response in the PUT endpoints

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala [hunk 1]
```diff
@@ -36,22 +36,22 @@ class SubscriptionFieldsService @Inject()(repository: SubscriptionFieldsReposito
...
 
     Logger.debug(s"[upsert subscription fields] clientId: $clientId, apiContext: $apiVersion, apiVersion: $apiVersion")
-
-    repository.fetch(clientId, apiContext, apiVersion) flatMap {
-      o =>
-        o.fold(
-          create() map { x => (x, true) }
-        )(
-          existing => update(existing.fieldsId) map { x => (x, false) }
-        )
```
Comment (googley42) [inline app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala]: I would recommend pushing this sort of logic into the Repository, and extracted if possible to the `MongoCrudHelper`. Also this logic may not be necessary as we get information about whether operation resulted in insert or update from the  `WriteResult`

File: app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala [hunk 2]
```diff
@@ -28,9 +28,23 @@ import scala.concurrent.Future
...
+  def upsert(apiContext: ApiContext, apiVersion: ApiVersion, fieldDefinitions: Seq[FieldDefinition]): Future[(FieldsDefinitionResponse, Boolean)] = {
+
+    def update(existingApiContext: String, existingApiVersion: String): Future[FieldsDefinitionResponse] =
+      save(FieldsDefinition(apiContext.value, apiVersion.value, fieldDefinitions))
+
+    def create(): Future[FieldsDefinitionResponse] =
+      save(FieldsDefinition(apiContext.value, apiVersion.value, fieldDefinitions))
+
     Logger.debug(s"[upsert fields definition] apiContext: $apiContext, apiVersion: $apiVersion, fieldDefinitions: $fieldDefinitions")
-    repository.save(FieldsDefinition(apiContext.value, apiVersion.value, fieldDefinitions))
```
Comment (googley42) [inline app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala]: See above comment re pushing this sort of logic into the Repository

**Comment by gokyo** [comment]
Merging this PR to `master`.
The improvement suggested by @googley42 will be addressed in a separate PR.

**Comment by googley42** [comment]
:+1: providing comments are looked at in next PR

**Pull request merged**